data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3ObjectHandler.java

```diff
@@ -5,11 +5,27 @@
 package org.opensearch.dataprepper.plugins.source;
 
 import java.io.IOException;
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSet;
 
 /**
  * A S3ObjectHandler interface must be extended/implement for S3 Object parsing
  *
  */
 public interface S3ObjectHandler {
+    /**
+     * Parse S3 object content using S3 object reference and pushing to buffer when acknowledgement set is present
+     * @param s3ObjectReference Contains bucket and s3 object details
+     * @param acknowledgementSet acknowledgement set for the object
+     *
+     * @throws IOException exception is thrown every time because this is not supported
+     */
+    void parseS3Object(final S3ObjectReference s3ObjectReference, final AcknowledgementSet acknowledgementSet) throws IOException;
+
+    /**
+     * Parse S3 object content using S3 object reference and pushing to buffer
+     * @param s3ObjectReference Contains bucket and s3 object details
+     *
+     * @throws IOException When Input/Output exception occur while processing the data.
+     */
     void parseS3Object(final S3ObjectReference s3ObjectReference) throws IOException;
```

Comment:
Do we need this method?

Reply:
I thought about keeping two API for clean interfacing. For now, I am deleting it. We can add it back, if needed.